source/os/include/threads.h

```diff
@@ -38,40 +39,77 @@ typedef unsigned long threadreturn;
 typedef threadreturn threadcall threadfunc(void*);
 #endif
 
-class atomic_int {
-#ifdef _MSC_VER
-	volatile long value;
-#else
-	volatile int value;
-#endif
+class atomic_int : public std::atomic<int32_t>
+{
 public:
-	int get_basic();
-	void set_basic(int val);
-
-	int operator++();
-	int operator++(int);
-	int operator--();
-	int operator--(int);
-	
-	int operator+=(int value);
-	int operator-=(int value);
-
-	int operator|=(int value);
-	int operator&=(int value);
+	// Offered for backwards compatibility
+	int32_t get_basic()
+	{
+		return load();
```

Comment:
Previously these provided non-atomic get and set. 

I didn't see how that would be valid, since it goes against the basic concept of std::atomic. So now they conduct the operations atomically.

Eventually these functions should go away, but they were used in a lot of places and I didn't see any harm in keeping them for now.